processing/src/main/java/io/druid/segment/ColumnSelectorFactory.java

```diff
@@ -27,6 +27,7 @@
 public interface ColumnSelectorFactory
 {
   public DimensionSelector makeDimensionSelector(DimensionSpec dimensionSpec);
+  public DimensionSelector makeDictEncodedStringDimensionSelector(DimensionSpec dimensionSpec);
```

Comment:
My intent here was:

- makeDimensionSelector() is a top level function that gets called by whatever is reading the Cursor
- makeDimensionSelector() calls handler.getDimensionSelector(this...), the handler controls whether it wants to call makeDictEncodedStringDimensionSelector or makeFloatColumnSelector, e.g. for a Float:
https://github.com/jon-wei/druid/blob/rebased_flex_dims2/processing/src/main/java/io/druid/segment/FloatDimensionHandler.java#L115
- I didn't introduce a new interface since the anonymous implementations of DimensionSelector returned have dependencies on properties of the Cursor, which differ between IncrementalIndex and QueryableIndex